fasta/assemblyStats.go

```diff
@@ -12,37 +12,48 @@ import (
 // AssemblyStats takes the path to a fasta file and a flag for whether lower case letters
 // should count as assembly gaps.  Five ints are returned, which encode:
 // the N50 size, half the size of the genome, size of the genome, size of the largest contig, and the number of contigs.
-func AssemblyStats(infile string, countLowerAsGaps bool) (int, int, int, int, int) {
+func AssemblyStats(infile string, countLowerAsGaps bool) (int, int, int, int, int, int) {
 	records := Read(infile)
-	var genomeLength int
 
 	contigList := MakeContigList(records, countLowerAsGaps)
 
-	for i := 0; i < len(contigList); i++ {
-		genomeLength += contigList[i]
-	}
+	genomeLength := calculateGenomeLength(contigList)
+
 	sort.Ints(contigList)
```

Comment:
contigList is sorted here when AssemblyStats is called, before CalculateN50L50 is called. Do you want to update the usage of CalculateN50L50 to make sure that if CalculateN50L50 is called outside of AssemblyStats, they know to sort their contigList beforehand?

Reply:
Yes I agree that it is important for people to know that. I believe it already say it: "`// CalculateN50L50 takes a sorted slice of contig lengths and the size of half the genome. It returns the N50 size and L50 size.`"